python/cuml/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -364,19 +364,15 @@ def partial_inference(self, X, delayed, **kwargs):
         partial_infs = self._partial_inference(
             X=X, op_type="regression", delayed=delayed, **kwargs
         )
-
-        def reduce(partial_infs, workers_weights, unique_classes=None):
-            regressions = dask.array.average(
-                partial_infs, axis=1, weights=workers_weights
-            )
-            merged_regressions = regressions.compute()
-            return merged_regressions
-
-        datatype = (
-            "daskArray" if isinstance(X, dask.array.Array) else "daskDataframe"
+        workers_weights = self._get_workers_weights()
+        merged_regressions = dask.array.average(
+            partial_infs, axis=1, weights=workers_weights
         )
 
-        return self.apply_reduction(reduce, partial_infs, datatype, delayed)
+        if delayed:
```

Comment:
I'm not sure how to interpret this keyword, since you're getting a dask array either way, but it's the same behavior as `self.apply_reduction`.

Reply:
I'm not exactly sure what you mean by "it's the same behavior as `self.apply_reduction`, but it seems to mean the opposite of `persist` ? @viclafargue could you provide more context here?

Reply:
I was just noting that it's the same behavior as we'd previously take [here](https://github.com/rapidsai/cuml/blob/0e033acc0316d7959d21f269d44559ced9865935/python/cuml/cuml/dask/ensemble/base.py#L370-L373), for any reviewers who might be confused about this like I was. Probably not worth changing any behavior in this PR.

Reply:
Sorry for the delayed response, @csadorf. It's been quite a while since I last worked on this, but the changes in this PR look very reasonable to me.